### ISSUE TITLE
Share button extremely slow to appear when split screen being used.

### ISSUE THREAD

**Issue by g1bwt** [mozilla-mobile/firefox-ios#16741]
## Steps to reproduce
Firefox 118.1 on iPadOS 17.0.3 on iPad Air 5th gen.

Open Firefox, landscape orientation. Navigate to a web page. Tap on the hamburger menu icon top right of the screen. The New tab menu will open. Tap on the Share icon and a further menu will open giving access to further actions such as Airdrop, print etc. Tap anywhere other than that menu to close it.

Now with problem.

Open Firefox, landscape oriented. Click on the 3 dots at top of screen and select “Split View”. Firefox will shift to the left on the display and a prompt will appear at the top screen asking choose another app. I use the Mastodon app Ice Cubes version 1.8.5. The two apps will appear side by side. Navigate to a web page. Tap on the hamburger menu icon now at bottom right of the screen. The New tab menu will open. Tap one the Share icon. Nothing happens. The menu that should open may open later, but not instantly like before it was split screen.

This has happened since at least Firefox IOS version 117. I have tried force quitting Firefox, removing and reinstalling Firefox from the App Store and power cycling the iPad, all to no avail.

The Share action works virtually instantly on other apps that I use. I don’t remember the problem on earlier versions of iPadOS.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-7533)

**Comment by data-sync-user** [comment]
➤ Daniela Arcese commented:

Andrei Bodea can you easily reproduce this?

**Comment by g1bwt** [comment]
I have completely erased and reinstalled everything on this iPad Air. It was installed from an iCloud backup. The problem is still there, unfortunately I didn’t check if it worked with the iPad setup for a new user, sorry.

One extra thing that I did notice was when I scrolled up on a web page, the Share menu briefly flashed up and then closed. I didn’t see this before.

**Comment by abodea** [comment]
Thank you for the response @g1bwt, and hello @dnarcese I will take a look and add my findings here asap.

**Comment by g1bwt** [comment]
> Thank you for the response @g1bwt can you please provide a video with this issue? 

The attached video should show the split screen, touching the hamburger menu and the menu appearing. I then touched the "Share" option and the menu disappears. I then scroll up the web page and, very briefly, the sharing options appear and the hide.

Thanks for looking at it.

https://github.com/mozilla-mobile/firefox-ios/assets/110241691/b5cb44ec-a90c-435a-8b57-c5386a32ad54

**Comment by data-sync-user** [comment]
➤ Norberto Andres Furlan commented:

Andrei Bodea can you send the steps to reproduce this issue?

**Comment by data-sync-user** [comment]
➤ Andrei Bodea commented:

Hello, Norberto Andres Furlansorry for the late response I was in PTO last week.

Here are my findings:

  * For the RC build from AppStore where I have the share button on the hamburger menu I tried with the exact steps but I was not able to reproduce this issue while being in split screen with iPad Air (iOS 15.7) with 120.1 (36298).
  * For the TF builds for example the latest v121 (36341) I have the share button only on the URL bar at the top and the button is no longer displayed on the hamburger menu, but I tried to reproduce with the exact steps > set the split screen and attempt to open the share menu. Everything worked as expected I had no issues opening that section.
I will try it more to reproduce it but so far no success on my end, I can see in the video attached that somehow when the reporter opens the share menu → nothing happens → when he opens the hamburger menu again the share menu is instantly displayed and then everything is dismissed.

Norberto Andres Furlan Daniela Arceseis there something you know about the share button being moved to the URL bar on the Prod but with it on the hamburger menu for the TF builds?

**Comment by dnarcese** [comment]
@g1bwt I have also not been able to reproduce. Are there any actions you have enabled in the share menu that you think could be causing this?

**Comment by g1bwt** [comment]
@dnarcese Many thanks to you all for testing my problem.
I have been using the Stage Manager setting lately (I prefer this workflow method over the split screen way), which didn’t have the problem.
I went back to split screen to test it and now with Firefox 120.3 (36471] the problem seems to have been resolved. When I tap on the Share icon, the sharing window comes up instantly. 
I was using Firefox 120.1 before, and also since I reported the problem, the iPad has been updated to 17.1.2 from 17.0.3. So maybe with these updates, the problem has been sorted and I would think that this bug report can be closed.

I am sorry that I didn’t test it earlier, but have been, and still am suffering, from an illness which meant I forgot to keep testing.
